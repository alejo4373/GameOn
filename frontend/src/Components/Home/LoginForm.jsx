import React, { Component } from 'react';
import axios from 'axios';

export default class LoginForm extends Component {
  state = {
    username: '',
    password: ''
  }

  handleTextInput = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit= e => {
    e.preventDefault();
    const { username, password } = this.state;

    axios
      .post("/login", {
        username: username,
        password: password
      })
      .then(res => {
        this.props.handleLoginResponse(null, res.data)
      })
      .catch(err => {
        this.props.handleLoginResponse(err, null)
      });
  };

  render() {
    const { username, password } = this.state;
    return (
      <div>
        <h1 className="form-title">Log In</h1>
          <form onSubmit={this.handleSubmit}>
            <input
              id="login-input"
              type="input"
              placeholder="Username"
              name="username"
              value={username}
              onChange={this.handleTextInput}
            />
            <br />
            <input
              id="login-password"
              type="password"
              placeholder="Password"
              name="password"
              value={password}
              onChange={this.handleTextInput}
            />
            <br />
            <input id="login-submit" type="submit" value="Submit" />
            <p>Don't have an Account <a onClick={this.props.toggleForm}>Sign Up</a></p>
          </form>
      </div>
    )
  }
}