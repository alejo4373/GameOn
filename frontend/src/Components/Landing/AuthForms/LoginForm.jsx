import React, { Component } from 'react';
import { FormGroup, FormControl, Form, Button, Alert } from 'react-bootstrap';
import { Link } from 'react-router-dom';

export default class LoginForm extends Component {
  state = {
    username: '',
    password: ''
  }

  handleTextInput = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleDemoLogin = () => {
    const credentials = {
        username: 'demoUser',
        password: 'demo'
    }
    this.props.loginUser(credentials)
  }

  handleSubmit = e => {
    e.preventDefault();
    this.props.loginUser(this.state)
  };

  render() {
    const { username, password } = this.state;
    const { msg } = this.props;
    return (
      <div>
        <h1 className="form-title">Log In</h1>
          <Form horizontal onSubmit={this.handleSubmit}>
            <FormGroup>
              <FormControl
                type="text"
                placeholder="Username"
                name="username"
                value={username}
                onChange={this.handleTextInput}
              />
            </FormGroup>
            <FormGroup>
              <FormControl
                type="password"
                placeholder="Password"
                name="password"
                value={password}
                onChange={this.handleTextInput}
              />
            </FormGroup>
            <div className='login-buttons'>
              <Button bsStyle='success' type="submit">Log In</Button>
              {' '}
              <Button bsStyle='info' onClick={this.handleDemoLogin}>Demo</Button>
            </div>
            <p>Don't have an Account <Link to='/signup'>Sign Up</Link></p>
          </Form>
          {msg ? <Alert bsStyle='danger'> {msg} </Alert> : null}
      </div>
    )
  }
}
