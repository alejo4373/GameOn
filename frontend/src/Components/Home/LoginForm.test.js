import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm';

describe('<LoginForm/>', () => {
  it('form renders as it did last time (snapshot)', () => {
  const component = renderer.create(<LoginForm/>)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('click to submit form', (done) => {
    const callback = (err, user) => {
      expect(err).toEqual(new Error("Network Error"))
      done();
    }

    const wrapper = mount(
      <LoginForm handleLoginResponse={callback}/> 
    );
    wrapper.find('form').simulate('submit');
  });

  it('has two input texts, one of them is type password');
  it('doesn\'t submit if either input field is empty' );
  it('has a submit button with a function attached to it');
});