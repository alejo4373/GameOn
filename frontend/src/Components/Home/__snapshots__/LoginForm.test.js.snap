// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm/> form renders as it did last time (snapshot) 1`] = `
<div>
  <h1
    className="form-title"
  >
    Log In
  </h1>
  <form
    className="form-horizontal"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control"
        id={undefined}
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        id={undefined}
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div
      className="login-buttons"
    >
      <button
        className="btn btn-success"
        disabled={false}
        type="submit"
      >
        Log In
      </button>
       
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Demo
      </button>
    </div>
    <p>
      Don't have an Account 
      <a
        onClick={undefined}
      >
        Sign Up
      </a>
    </p>
  </form>
</div>
`;
